src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1383,6 +1384,174 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
+
+        then: "CHeck excess rules are not registered on device"
+        wait(WAIT_OFFSET) {
+            { verifySwitchRules(sw.dpId) }
+        }
+
+
+        cleanup: "Remove created flow and registered devices, revert switch props"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        database.removeConnectedDevices(sw.dpId)
+        initialProps && switchHelper.updateSwitchProperties(sw, initialProps)
+    }
+
+    @Tidy
+    def "System starts detect connected after device properties turned on"() {
+        given: "A switch with devices feature turned off"
+        def tgService = traffExamProvider.get()
+        assumeTrue(topology.activeTraffGens.size() > 0, "Require at least 1 switch with connected traffgen")
+        def sw = topology.activeTraffGens[0].switchConnected
+        def initialPropsSource = enableMultiTableIfNeeded(true, sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def initialPropsDst = enableMultiTableIfNeeded(true, dst.dpId)
+        def flow = flowHelperV2.randomFlow(sw, dst)
+
+        def outerVlan = 100
+        flow.source.vlanId = outerVlan
+        flow.source.innerVlanId = 200
+
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+        northboundV2.getConnectedDevices(flow.source.switchId).ports.empty
+        northboundV2.getConnectedDevices(flow.destination.switchId).ports.empty
+
+        when: "update device properties and send lldp and arp packets on src flow endpoint and match outerVlan only"
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = true
+            it.switchArp = true
+        })
+
+        def srcLldpData = LldpData.buildRandom()
+        def dstLldpData = LldpData.buildRandom()
+        def srcArpData = ArpData.buildRandom()
+        def dstArpData = ArpData.buildRandom()
+        [[flow.source, srcLldpData, srcArpData], [flow.destination, dstLldpData, dstArpData]].each {
+            endpoint, lldpData, arpData ->
+                new ConnectedDevice(tgService, topology.getActiveTraffGen(endpoint.switchId), [outerVlan]).withCloseable {
+                    it.sendLldp(lldpData)
+                    it.sendArp(arpData)
+                }
+        }
+
+        then: "Getting connecting devices doesn't show corresponding devices on src endpoint"
+        def response = northboundV2.getConnectedDevices(sw.dpId)
+        sleep(3000)
+        Wrappers.timedLoop(3) {
```

Comment:
Wrong sleep usage here. We use 'sleep' or 'timedLoop' to check that something has NOT happened. When we expect something to appear, we should use 'wait'
```
Wrapper.wait(whateverTimeout) {
    //check that device appeared
}
```

In case of wait we will terminate loop and continue as soon as our verification is satisfied. In case of sleep+timedLoop it will be a constant 6 seconds delay in this test, which should be avoided

Reply:
removed